central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,240 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	pgStore "github.com/stackrox/rox/central/auth/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestAuthDatastorePostgres(t *testing.T) {
+	suite.Run(t, new(datastorePostgresTestSuite))
+}
+
+type datastorePostgresTestSuite struct {
+	suite.Suite
+
+	ctx       context.Context
+	pool      *pgtest.TestPostgres
+	store     pgStore.Store
+	datastore DataStore
+}
+
+func (s *datastorePostgresTestSuite) SetupTest() {
+	s.ctx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Access),
+		),
+	)
+
+	s.pool = pgtest.ForT(s.T())
+	s.Require().NotNil(s.pool)
+
+	s.store = pgStore.New(s.pool.DB)
+	s.datastore = New(s.store)
+}
+
+func (s *datastorePostgresTestSuite) TearDownTest() {
+	s.pool.Teardown(s.T())
+	s.pool.Close()
+}
+
+func (s *datastorePostgresTestSuite) TestAddConfig() {
+	testCases := []struct {
+		config *storage.AuthMachineToMachineConfig
+		err    error
+	}{
+		{
+			config: &storage.AuthMachineToMachineConfig{},
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: nil,
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				Id: "some-id",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1h",
+			},
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "24h1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1h",
+				Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+			},
+			err: errox.InvalidArgs,
+		},
```

Comment:
A contrary case may be also useful: Type is GITHUB_ACTIONS and Generic IssuerConfig is provided.